### PR TITLE
TropicalGeometry: fixed bug in binomial tropical varieties

### PR DESCRIPTION
**Pull request by YueRen**
Fixed two small bugs in the code, found together with @oskarhenriksson:
- redundant GB computation (generator suffices)
- product of the (non-zero) diagonal entries errors when matrix has more rows than cols (e.g., when the generating set is not minimal)

### CODE REVIEW

File: src/TropicalGeometry/variety.jl [hunk 1]
```diff
@@ -243,19 +243,20 @@ end
...
     b = [ QQ(nu(last(collect(coeff))/first(collect(coeff)))) for coeff in coefficients.(G)]
 
     ###
     # Compute tropical variety multiplicity
     ###
-    @req ncols(A)>nrows(A) "input needs to be a GB"
-    weight = abs(prod([snf(A)[i,i] for i in 1:nrows(A)]))
+    # Todo: replace lines below by `diagonal` when https://github.com/Nemocas/Nemo.jl/pull/1627 is merged
+    snfA = snf(A)
+    snfAdiag = [snfA[i,i] for i in 1:min(nrows(A),ncols(A))]
```
Comment (thofma) [inline src/TropicalGeometry/variety.jl:258]: snfAdiag = elementary_divisors(A)

Comment (thofma) [inline src/TropicalGeometry/variety.jl:258]:
```suggestion
    snfAdiag = elementary_divisors(A)
```

Comment (thofma) [inline src/TropicalGeometry/variety.jl]:
```suggestion
    snfAdiag = elementary_divisors(A)
```
Sorry, now I have it.

Comment (YueRen) [reply src/TropicalGeometry/variety.jl]: Thanks for the tip with `elementary_divisors`, this will come in handy in multiple places of my code!

**Comment by fingolfin** [comment]
Failure in the documentation CI test?

Can anyone review this?

**Comment by benlorenz** [comment]
The docs error should be resolved with #3214 and should disappear after this re-open.

**Comment by YueRen** [comment]
If all tests are passed, can somebody please merge this? @oskarhenriksson is visiting me today, and I'd prefer if we can work on the latest master branch.

**Review by benlorenz — APPROVED** [review]

**Pull request merged**